nifi-commons/nifi-utils/src/main/java/org/apache/nifi/util/file/FileUtils.java

```diff
@@ -575,11 +577,43 @@ public static long getContainerCapacity(final Path path) {
 
     /**
      * Returns the free capacity for a given path
+     *
      * @param path path
      * @return usable space
      */
     public static long getContainerUsableSpace(final Path path) {
         return path.toFile().getUsableSpace();
     }
 
+    /**
+     * Returns the size for a given directory.
+     * @param path directory path
+     * @param logger logger
+     * @return the size in bytes
+     */
+    public static long getDirectorySize(Path path, final Logger logger) {
+        long size = 0;
+        try (Stream<Path> walk = Files.walk(path)) {
+            size = walk
+                    .filter(Files::isRegularFile)
+                    .mapToLong(getFileSizeByPathFunction(logger))
+                    .sum();
+
+        } catch (IOException e) {
+            logger.error("IO exception occured while tried to get the size of the directory {}", path, e);
+        }
+        return size;
+    }
+
+    private static ToLongFunction<Path> getFileSizeByPathFunction(Logger logger) {
+        return p -> {
+            try {
+                return Files.size(p);
+            } catch (IOException e) {
+                logger.error("Failed to get size of directory {}", p, e);
+                return 0L;
+            }
+        };
+    }
```

Comment:
Unnecessarily confusing abbreviation. Also this is called with regular files, not directories.
```suggestion
        return path -> {
            try {
                return Files.size(path);
            } catch (IOException e) {
                logger.error("Failed to get size of file {}", path, e);
                return 0L;
            }
        };
```